test/benchmark_http_client_test.cc

```diff
@@ -37,37 +38,81 @@ using namespace testing;
 
 namespace Nighthawk {
 
-class BenchmarkClientTestBase : public Envoy::BaseIntegrationTest,
-                                public TestWithParam<Envoy::Network::Address::IpVersion> {
+class PythonIntegrationTestBase : public TestWithParam<Envoy::Network::Address::IpVersion> {
 public:
-  BenchmarkClientTestBase()
-      : Envoy::BaseIntegrationTest(GetParam(), realTime(), BenchmarkClientTestBase::envoy_config),
-        api_(thread_factory_, store_, timeSystem(), file_system_),
-        dispatcher_(api_.allocateDispatcher()) {}
+  PythonIntegrationTestBase(const Envoy::Network::Address::IpVersion version)
+      : version_(version), pipe_(nullptr){};
+
+  void startPythonIntegrationWrapper(const bool use_https) {
+    std::string args;
 
-  static void SetUpTestCase() {
-    Envoy::Filesystem::InstanceImplPosix file_system;
-    envoy_config = file_system.fileReadToEnd(Envoy::TestEnvironment::runfilesPath(
-        "test/test_data/benchmark_http_client_test_envoy.yaml"));
-    envoy_config = Envoy::TestEnvironment::substitute(envoy_config);
+    if (GetParam() == Envoy::Network::Address::IpVersion::v4) {
+      args.append(" ipv4");
+    } else {
+      args.append(" ipv6");
+    }
+    args.append(" http");
+    if (use_https) {
+      args.append("s");
+    }
+    pipe_ = popen(Envoy::TestEnvironment::runfilesPath("integration/cpp_benchmark_client_server")
+                      .append(args)
+                      .c_str(),
+                  "r");
+    RELEASE_ASSERT(pipe_ != nullptr, "Failed to open pipe");
```

Comment:
So, we have C++ calling Python calling C++ here? How come we can't just launch the test server directly as a forked process?